kong/router/compat.lua

```diff
@@ -249,27 +252,34 @@ local function get_priority(route)
     end
   end
 
-  local max_uri_length = 0
+  local uri_length = 0
   local regex_url = false
 
   if not is_empty_field(paths) then
     match_weight = match_weight + 1
 
-    for _, p in ipairs(paths) do
-      if is_regex_magic(p) then
-        regex_url = true
-
+    for index, p in ipairs(paths) do
```

Comment:
if we are enforcing one path per route, can we just assert that `#route.paths == 1` here and avoid this loop? 

Or somewhere in atc.lua this gets called with a route that can have many paths?

Reply:
You are right.  I kind of expect that someone points out the possibility to group regular expressions and paths of equal lengths, so I left the loop in place in preparation of that.